### PR TITLE
Allow YouTube playing different projection types of VR videos continuously

### PR DESCRIPTION
**Pull request by HollowMan6**
Since YouTube plugin infers video projection types according to words in video title and description, and if `mozVideoProjection` is already there in the url, the plugin will respect the value and won't infer any more.

This works perfectly fine for playing a single video, but if we try to continue playing other projection types of video without leaving the Youtube, the inferred `mozVideoProjection` will get preserved in the url, which makes all the videos continue using the same projection type.

As a result, we should reset the page projection after navigating to a new page that is different from initial page.

Also we should wait for yt-update-title event to happen so that we can get the updated title and description for projection inferring:
https://www.sitepoint.com/community/t/how-to-get-the-new-youtube-title-when-yt-navigate-finish-runs/377799/2

Remove the window load listener as this is not neccessary, youtube events is enough.

To reproduce:
1. Go to https://www.youtube.com/results?search_query=3d+top+bottom
2. Click into a 3d top bottom format video to play, e.g. the first one:
com igalia wolvic-20230929-225930
3. Click the back button on the controller to go back
4. Click into a 3d sbs format video to play, e.g. the one here:
com igalia wolvic-20230929-230004

You will find that 3d sbs format video's projection format is not correctly recognized while we can actually play that correctly if being played separately.

### CODE REVIEW

File: app/src/main/assets/extensions/fxr_youtube/main.js [hunk 1]
```diff
@@ -133,7 +134,7 @@ class YoutubeExtension {
             return; // Only override projection in the Youtube watching page.
         }
         const qs = new URLSearchParams(window.location.search);
-        if (qs.get(VIDEO_PROJECTION_PARAM)) {
+        if (qs.get(VIDEO_PROJECTION_PARAM) && !qs.get(VIDEO_PROJECTION_INFERRED_PARAM)) {
```
Comment (svillar) [inline app/src/main/assets/extensions/fxr_youtube/main.js:137]: What about registering a navigation handler which removes the params on navigation forcing us to recompute the projection?

Comment (HollowMan6) [reply app/src/main/assets/extensions/fxr_youtube/main.js:137]: Okay, let me try. Not sure if that way will 100% work

**Review by svillar — CHANGES_REQUESTED** [review]
The idea is good but it isn't working for me. Using the same search you used as an example I played https://www.youtube.com/watch?v=OBkoJyU6124 first and then tried with  https://www.youtube.com/watch?v=u0hmMt5znX0 and the URL didn't change

Bished: I'll call the parametter forceUpdate or something including "force" at least

**Comment by HollowMan6** [comment]
> The idea is good but it isn't working for me. Using the same search you used as an example I played https://www.youtube.com/watch?v=OBkoJyU6124 first and then tried with https://www.youtube.com/watch?v=u0hmMt5znX0 and the URL didn't change
> 
> Bikeshedding: I'd call the parametter forceUpdate or something including "force" at least

Looks like Gecko only has a popstate event: https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event There's no pushstate event now, that's why it's not working. So here we have no choice but to listen to the path changes, and we have to delay the video projection update as the video title/description is not available immediately.

**Comment by svillar** [comment]
What about replacing `pushstate` event by `hashchange`. Would that work?

**Comment by svillar** [comment]
> > What about replacing `pushstate` event by `hashchange`. Would that work?
> 
> No, the fragment identifier is used nowhere here. And I think the current way is good enough.

It should be simpler.

In understand that popstate is not enough because we don't get history changes when the YT player goes to the next video. So the question is why not using the [YT player API](https://developers.google.com/youtube/iframe_api_reference) for that? idk maybe use `onReady` ?

**Comment by HollowMan6** [comment]
> It should be simpler.
> 
> In understand that popstate is not enough because we don't get history changes when the YT player goes to the next video. So the question is why not using the [YT player API](https://developers.google.com/youtube/iframe_api_reference) for that? idk maybe use `onReady` ?

Updated with a simpler version. The link you provided is the iframe API for building the website embedded with YouTube videos, so it's not useful here. Then I finally found some events that YouTube player will fire when they make updates, and get rid of those path observation codes.

**Review by svillar — APPROVED** [review]
Seems to work fine. Thanks!

**Pull request merged**